### ISSUE TITLE
BufferWhileBy should flush source stream on falsy events from "otherObs"

### ISSUE THREAD

**Issue by shamansir** [rpominov/kefir#116]
Instead of flushing on the next event from the source stream. So it should rather be like:

```javascript
var foo = Kefir.sequentially(100, [1, 2, 3, 4, 5, 6, 7, 8]);
var bar = Kefir.sequentially(200, [false, true, false]).delay(40);
var result = foo.bufferWhileBy(bar);
result.log();
```

```
> [sequentially.bufferWhileBy] <value> [1, 2]
> [sequentially.bufferWhileBy] <value> [3]
> [sequentially.bufferWhileBy] <value> [4]
> [sequentially.bufferWhileBy] <value> [5, 6]
> [sequentially.bufferWhileBy] <value> [7]
> [sequentially.bufferWhileBy] <value> [8]
> [sequentially.bufferWhileBy] <end>
```

```
foo:     ----1----2----3----4----5----6----7----8X
bar:     -----------f---------t---------fX

result:  -----------•--•----•-----------•--•----•X
               [1, 2] [3]  [4]     [5, 6] [7]  [8]
```

**Comment by rpominov** [comment]
I've added an option for this. Will it solve your problem? Also can you try it out before we release (it in master now)?

**Comment by shamansir** [comment]
[Here's my implementation](https://github.com/rpominov/kefir/compare/96c60a81d4039314ec20f9061a5a11a9c80568a2...shamansir:master) where it is flushed this way by default. I'll try your version in few minutes, thank you!

**Comment by shamansir** [comment]
Yes, it works the same way with even fewer changes, thank you!

**Comment by rpominov** [comment]
Released in `2.7.0`
